src/analyses/variable-sensitivity/abstract_object.cpp

```diff
@@ -172,6 +172,20 @@ exprt abstract_objectt::to_constant() const
   return nil_exprt();
 }
 
+exprt abstract_objectt::to_predicate(const exprt &name) const
+{
+  if(is_top())
+    return true_exprt();
+  if(is_bottom())
+    return false_exprt();
+  return to_predicate_internal(name);
+}
+
+exprt abstract_objectt::to_predicate_internal(const exprt &name) const
+{
+  return nil_exprt();
```

Comment:
As I understand it, this should never be directly called. right?  It is here because it is overloaded?  (I guess this is a point where you are right about the separation of interface and base implementation. )  If so, `UNREACHABLE` might be more appropriate.